server/hl7/hl7.controller.js

```diff
@@ -108,5 +109,63 @@ function getUserFiles(req, res, next) {
   return next(err);
 }
 
+function getMessageByIndexOrId(req, res, next) {
```

Comment:
Not sure how I feel about one endpoint serving multiple functions (byIndex vs byId) typical REST would say this endpoint would be byId, but I think byIndex is more useful for us. This is good, just pontificating

Reply:
I knew this was bad when I did it lol- fixing it by separating into two separate REST endpoints with one helper function